### DISCUSSION TITLE
public model

### DISCUSSION THREAD

**Discussion by dengsong666** [prisma/prisma#23540] [Q&A] [answered]
### Question

In prisma, I want to extract id, createAt, updateAt as a public model. What do I do?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @dengsong666 👋 

Currently, Prisma ORM does not support model extensions or abstract models that can be extended by other models. This means you cannot extract `id`, `createdAt`, and `updatedAt` into a public model and extend it in other models. Each model in Prisma needs to define its own fields.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
